igcollect/ping.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+"""igcollect - Latency collection
+Copyright (c) 2020 InnoGames GmbH
+"""
+
+# This script is used to collect ping data to evaluate availability and latency
+# of our network and make it easy to create a Graphite Dashboard
+#
+# Please note that the package fping has to be installed on your Linux server!
+
+
+import argparse
+import subprocess
+import time
+import socket
+
+from multiprocessing import Pool
+from os import cpu_count
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Ping collector script for Graphite ')
+    parser.add_argument('hosts', nargs='+',
+                        help='the hosts to ping')
+    parser.add_argument('-c', '--count', type=int, default=20, 
+                        choices=range(1,60),
+                        help='how many times the script will try to ping '
+                             '(number has to be bigger '
+                             'than 1 and smaller than 60) (default= 20)')
+    parser.add_argument('-p', '--prefix',
+                        default='servers.{}.system.ping'.format(
+                            socket.gethostname().replace('.', '_')),
+                        help='the path to the value in Graphite '
+                             '(default= servers.(hostname).system.ping)')
+    parser.add_argument('-t', '--timeout', type=int, default=300,
+                        help='time in milliseconds '
+                             'till the timeout is retched (default= 300)')
```

Comment:
https://en.wiktionary.org/wiki/retch#English

Description of timeout here differs from the one in code below, where it's described as delay and it does not explain at all what is it for.

Reply:
these are different values !